File: template/c/tests/test_add.c
#include <criterion/criterion.h>

#include "add.h"

Test(add, positive) { cr_assert_eq(add(2, 3), 5); }

Test(add, negative) { cr_assert_eq(add(-2, -3), -5); }

Test(add, zero) { cr_assert_eq(add(0, 0), 0); }

File: template/ocaml/test/test_add.ml
let test_positive () = Alcotest.(check int) "2 + 3" 5 (App.Calc.add 2 3)

let test_negative () =
  Alcotest.(check int) "-2 + -3" (-5) (App.Calc.add (-2) (-3))

let test_zero () = Alcotest.(check int) "0 + 0" 0 (App.Calc.add 0 0)

let () =
  Alcotest.run "app"
    [
      ( "add",
        [
          Alcotest.test_case "positive" `Quick test_positive;
          Alcotest.test_case "negative" `Quick test_negative;
          Alcotest.test_case "zero" `Quick test_zero;
        ] );
    ]
